src/Essentials/src/Platform/WindowStateManager.ios.cs

```diff
@@ -101,9 +101,18 @@ class WindowStateManagerImplementation : IWindowStateManager
 			// if we have scene support, use that
 			if (OperatingSystem.IsIOSVersionAtLeast(13) || OperatingSystem.IsMacCatalystVersionAtLeast(13))
 			{
-				var scenes = UIApplication.SharedApplication.ConnectedScenes;
-				var windowScene = scenes.ToArray<UIWindowScene>().FirstOrDefault();
-				return windowScene?.Windows.FirstOrDefault();
+				try
+				{
+					using var scenes = UIApplication.SharedApplication.ConnectedScenes;
```

Comment:
First level of attack, make sure to dispose the "other" set so that there is no managed object around to get a miss-type.